app/common/gutil.ts

```diff
@@ -55,6 +55,11 @@ export function capitalizeFirstWord(str: string): string {
   return str.replace(/\b[a-z]/i, c => c.toUpperCase());
 }
 
+// Remove diacritics (accents and other signs).
+export function removeDiacritics(text: string): string {
```

Comment:
What do you think about using some external, widely used library here? I'm not an expert on this topic, but I don't think that `normalize` is removing diacritics. It didn't work for most of my tests.

Reply:
Isn't it the `.replace(/[\u0300-\u036f]` part that removes the diacritics? With the `normalize` part just shuffling the representation to encode them in a favorable way?

Reply:
Yes, the normalization separates the diacritics from the base characters, then the replace removes them. But I agree this is not obvious so I’ll look into making this more explicit. I would have liked to add unit tests too, but I was not sure where they should go. 🤔

Reply:
I'd like to know your opinion on using an external library. There are some advantages that I see:
- There are diacritics and diacritic letters; I think that for regular users there is no difference between them; for example: "łø" which from my perspective, should also be searchable by "lo".
- This `normalized` method can also be used for searching, and other autocomplete features. This would be a very useful and simple improvement. Other spreadsheets apps seem to work this way (though the implementation is probably different).
- External library probably supports more cases, and we don't need to write tests for it :).

Reply:
I’m not familiar with the JS libraries in this space, so I’ll do some research, but suggestions are welcome!

Reply:
It turns out that we already have something: https://lodash.com/docs/4.17.15#deburr

Reply:
There is also a problem with highlighting as the function `highlightMatches` works on "dirty" text, so it won't underline matched parts. I think we need to normalize the text on input and then somehow replace the output of that function with original characters (which won't be easy).  From brief research, I found this [topic](https://stackoverflow.com/questions/10758913/unicode-string-with-diacritics-split-by-chars) that might help. Some dirty implementation [example](https://gist.github.com/berhalak/ba0dfba0819a09ef9ee312f6701bbf9b):

Reply:
>  I would have liked to add unit tests too, but I was not sure where they should go.

Just to be clear about this, @berhalak added some tests related to Choice Lists, References, and Reference Lists here https://github.com/gristlabs/grist-core/pull/241 - if you rebase, they would be good places to add tests. We have a lot of internal tests that we've been cleaning up and adding to grist-core as they are needed (they sometimes need disentangling from internal cruft).